release/pkg/assets/assets.go

```diff
@@ -0,0 +1,323 @@
+package assets
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	assetconfig "github.com/aws/eks-anywhere/release/pkg/assets/config"
+	assettypes "github.com/aws/eks-anywhere/release/pkg/assets/types"
+	"github.com/aws/eks-anywhere/release/pkg/filereader"
+	"github.com/aws/eks-anywhere/release/pkg/git"
+	"github.com/aws/eks-anywhere/release/pkg/images"
+	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	"github.com/aws/eks-anywhere/release/pkg/utils"
+)
+
+func getAssetsFromConfig(ac *assettypes.AssetConfig, rc *releasetypes.ReleaseConfig, eksDReleaseChannel, eksDReleaseNumber, kubeVersion string) ([]releasetypes.Artifact, error) {
```

Comment:
This is a 270 line function with 3 major for loops. At a glance its doing lots of different things, do you think we could break this down _and_ write unit tests.

Reply:
Will do